build-system/tasks/bundle-size/index.js

```diff
@@ -53,23 +57,42 @@ const bundleSizeAppBaseUrl = 'https://amp-bundle-size-bot.appspot.com/v0/';
  * @return {Map<string, number>} the bundle size in KB rounded to 2 decimal
  *   points.
  */
-function getBrotliBundleSizes() {
+async function getBrotliBundleSizes() {
   // Brotli compressed size fluctuates because of changes in the RTV number, so
   // normalize this across pull requests by replacing that RTV with a constant.
   const bundleSizes = {};
 
   log(cyan('brotli'), 'bundle sizes are:');
-  for (const filePath of globby.sync(fileGlobs)) {
-    const normalizedFileContents = fs
-      .readFileSync(filePath, 'utf8')
-      .replace(new RegExp(internalRuntimeVersion, 'g'), normalizedRtvNumber);
-
-    const relativeFilePath = path.relative('.', filePath);
-    const bundleSize = parseFloat(
-      (brotliSize.sync(normalizedFileContents) / 1024).toFixed(2)
+  const values = report(process.cwd(), content =>
+    content.replace(
+      new RegExp(internalRuntimeVersion, 'g'),
+      normalizedRtvNumber
+    )
+  );
+  let next = await values.next();
+  let pathMap = undefined;
+  if (!next.done) {
+    const compressed = next.value[0];
+    pathMap = new Map(
+      Array.from(compressed.keys()).map(path => [path, undefined])
     );
-    log(' ', cyan(relativeFilePath) + ':', green(`${bundleSize}KB`));
-    bundleSizes[relativeFilePath] = bundleSize;
+  }
+  while (!next.done) {
+    const {value: currentValue} = next;
+    pathMap.forEach((pathValue, filePath) => {
+      const retrievedValues = currentValue[0].get(filePath);
+      if (pathValue === undefined && retrievedValues) {
+        const brotliSize = retrievedValues[0][0];
+        if (brotliSize !== null) {
+          const relativePath = path.relative('.', filePath);
+          const reportedSize = parseFloat((brotliSize / 1024).toFixed(2));
+          log(' ', cyan(relativePath) + ':', green(reportedSize + 'KB'));
+          pathMap.set(path, brotliSize);
+          bundleSize[relativePath] = reportedSize;
```

Comment:
Is this a typo? `%s/bundleSize/bundleSizes/`

Reply:
It was indeed! Cannot wait for this project to move to TypeScript and help catch these!

Reply:
I would love to drive a TS migration